node/src/service/p2p/bootstrap.rs

```diff
@@ -61,46 +105,197 @@ impl BootstrapConfig {
+                    info!("Disconnected from {}", peer_id);
+                }
+                SwarmEvent::Behaviour(BootstrapBehaviourEvent::Rendezvous(event)) => on_rendezvous_event(&mut swarm, event, &mut registrations),
+                SwarmEvent::Behaviour(BootstrapBehaviourEvent::Gossipsub(event)) => on_gossipsub_event(event, *swarm.local_peer_id(), &mut registrations),
+                SwarmEvent::Behaviour(BootstrapBehaviourEvent::RequestResponse(event)) => on_request_response_event(&mut swarm, event, &registrations),
+                other => debug!("Encountered event: {other:?}"),
             }
-            SwarmEvent::Behaviour(BootstrapBehaviourEvent::Rendezvous(
-                rendezvous::server::Event::PeerRegistered { peer, registration },
-            )) => {
-                info!(
-                    "Peer {} registered for namespace '{}' for {} seconds",
-                    peer, registration.namespace, registration.ttl
+        }
+    }
+}
+
+/// Handles events within the rendezvous protocol
+fn on_rendezvous_event(
+    swarm: &mut Swarm<BootstrapBehaviour>,
+    event: rendezvous::server::Event,
+    registrations: &mut HashMap<PeerId, PeerInfo>,
+) {
+    match event {
+        rendezvous::server::Event::RegistrationExpired(registration) => {
+            let id = registration.record.peer_id();
+            info!(
+                "Registration for peer {} expired in namespace {}",
+                id, registration.namespace
+            );
+            // Registration expired, remove entry from hashmap
+            if registrations.remove(&id).is_none() {
+                error!(
+                    "Could not remove registration for {:?} because it was not found",
+                    id
                 );
             }
-            SwarmEvent::Behaviour(BootstrapBehaviourEvent::Rendezvous(
-                rendezvous::server::Event::DiscoverServed {
-                    enquirer,
-                    registrations,
-                },
-            )) => {
-                if !registrations.is_empty() {
-                    info!(
-                        "Served peer {} with {} new registrations",
-                        enquirer,
-                        registrations.len()
-                    );
+        }
+        rendezvous::server::Event::PeerRegistered { peer, registration } => {
+            info!(
+                "Peer {} registered for namespace '{}' for {} seconds",
+                peer, registration.namespace, registration.ttl
+            );
+            let peer_info = PeerInfo {
+                peer_id: peer,
+                multiaddrs: registration.record.addresses().to_vec(),
+            };
+            // Serialize PeerInfo
+            let encoded_peer_info = match bincode::serialize(&peer_info) {
+                Ok(info) => info,
+                Err(..) => {
+                    error!("Failed to serialize peer_info");
```

Comment:
I think this should never happen. But we should still add the info to the message if printing a log.

```suggestion
                    error!(?peer_info, "Failed to serialize peer_info");
```

Reply:
It woks a bit different in the [log library](https://docs.rs/log/latest/log/kv/index.html) but I understand your point. Will add the info to the log, same for the de-serialization.